.github/workflows/reusable-coveralls.yml

```diff
@@ -15,6 +15,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Can you explain why `fetch-depth: 0` is needed?

It gets all commits, so it takes long time. 
We should not use it as possible.

Reply:
I'm not sure what happened to composer, but it could no longer resolve `codeigniter4/codeigniter4` as the same as `codeigniter4/framework`. Upon checking, it seems composer needs to have the develop branch also checked out. By default, Github only checks out the head branch on PRs.

Reply:
I have pushed a different approach by fetching the base branch alongside the PR branch.

Reply:
Do the actions check the PR branch?

It seems this flow checks the base branch? Or am I wrong?
"Checkout": 84f02572ca6fa7aae75ab5026229bd3fa0a8d466
"Checkout base branch for PR": 0b0126cf8d7b00031832d6b097fe82ca8763897d
https://github.com/codeigniter4/CodeIgniter4/actions/runs/12281147091/job/34269293548?pr=9299

Reply:
The action checkouts the PR branch in a detached HEAD state, which I think is the cause of the merge commit you're seeing, i.e. `84f0257`.

Reply:
**Checkout** checks out `84f0257`.
And **Checkout base branch for PR** checks out `0b0126c`.
So **Run lint** checks `0b0126c` (`develop`)?

Reply:
ahh, I get what you mean. maybe that's the reason why it's not picking up the latest dependencies. I'll investigate later

Reply:
I have swapped the checkout with the checkout of the base branch being the first one.